### PR TITLE
feat: Readable diffs

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
When a `std::assert_eq!()` fails, it can be quite hard to see the difference.

# Changes
* Use `pretty_assertions::assert_eq()` instead of `std::assert_eq()`

# Tests
* Library security and code quality:  I had a quick look through the library and its dependencies.  It has few dependencies (2) and both dependencies are minimal.
* Licencing: The license is the standard BSD/apache dual license used in most Rust projects.
* `pretty_assertions` is already used extensively in the `dfinity/ic` repository.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: clippy.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@ disallowed-macros = [
     "std::println",
     "std::eprint",
     "std::eprintln",
+    "std::assert_eq"
```
[the remark this answers is not in the document]
Comment (daniel-wong-dfinity-org) [reply clippy.toml:6]: Maybe, mention this in a comment above `disallowed-macros`? I might say something like "There are (better) versions of these. This makes sure that we do not use the vanilla ones.".

**Review by daniel-wong-dfinity-org — APPROVED** [review]

File: clippy.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bitdivine) [reply clippy.toml:6]: I searched and found out that `disallowed-macros` supports giving a reason for each prohibition.  See: https://rust-lang.github.io/rust-clippy/master/index.html#/disallowed_macros

I tested and can confirm that the updated code detects prohibited macros in main (non-test) code but unfortunately the error message isn't printed along with the error:
```
error: use of a disallowed macro `std::assert_eq`
  --> rs/backend/src/accounts_store/schema.rs:86:9
   |
86 |         assert_eq!(1,2);
   |         ^^^^^^^^^^^^^^^
   |
   = note: Use pretty_assertions::assert_eq instead (from clippy.toml)
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#disallowed_macros
   = note: `-D clippy::disallowed-macros` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::disallowed_macros)]`

error: could not compile `nns-dapp` (lib) due to 1 previous error
```

[the remark this answers is not in the document]
Comment (bitdivine) [reply clippy.toml]: I looked into the ic-repo.  That already uses the reason field for disallowed methods.  It has no disallowed macros.

**Comment by bitdivine** [comment]
Re CHANGELOG - I guess the NNS team is free to update the rules but hitherto we have also recorded changes that are not user-visible.  For example, there is a section in the changelog on operations, which typically includes changes to CI, which are _usually_ not visible to users.  If it turned out that `pretty_assertions` is broken in some way for some of our use cases, developers would typically look in the changelog and git log for reasons why stuff broke, which is why I included it.  If a PR is so minor that no-one will ever go looking for it I typically omit an entry. :-) But you are of course free to decide as a team what you would like the rules to be.

File: clippy.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bitdivine) [reply clippy.toml]: The docs say that the reason SHOULD be included in the diagnostic.  Maybe we need a newer version of clippy for that to work.

[the remark this answers is not in the document]
Comment (bitdivine) [reply clippy.toml]: Resolving so that I can merge but happy to discuss further.

**Pull request merged**